api/apps/sdk_app.py

```diff
@@ -0,0 +1,204 @@
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+import json
+import os
+import re
+from datetime import datetime, timedelta
+from flask import request, Response
+from flask_login import login_required, current_user
+
+from api.db import FileType, ParserType, FileSource, StatusEnum
+from api.db.db_models import APIToken, API4Conversation, Task, File
+from api.db.services import duplicate_name
+from api.db.services.api_service import APITokenService, API4ConversationService
+from api.db.services.dialog_service import DialogService, chat
+from api.db.services.document_service import DocumentService
+from api.db.services.file2document_service import File2DocumentService
+from api.db.services.file_service import FileService
+from api.db.services.knowledgebase_service import KnowledgebaseService
+from api.db.services.task_service import queue_tasks, TaskService
+from api.db.services.user_service import UserTenantService, TenantService
+from api.settings import RetCode, retrievaler
+from api.utils import get_uuid, current_timestamp, datetime_format
+from api.utils.api_utils import server_error_response, get_data_error_result, get_json_result, validate_request
+from itsdangerous import URLSafeTimedSerializer
+
+from api.utils.file_utils import filename_type, thumbnail
+from rag.utils.minio_conn import MINIO
+
+LIMIT_LENGTH_OF_NAME = 2 ** 10
+
+
+# helper method to check whether it contains space
+def contains_space(s):
+    return any(char.isspace() for char in s)
+
+
+# ------------------------------ create a dataset ---------------------------------------
+@manager.route('/dataset', methods=['POST'])
+@validate_request("name")  # check name key
+def create():
+    # Check if Authorization header is present
+    if 'Authorization' not in request.headers:
+        return get_json_result(data=False, retmsg="Authorization header is missing!",
+                               retcode=RetCode.AUTHENTICATION_ERROR)
+
+    # Extract token from Authorization header
+    try:
+        token = request.headers.get('Authorization').split()[1]
+    except IndexError:
+        return get_json_result(data=False, retmsg="Token is missing!", retcode=RetCode.AUTHENTICATION_ERROR)
+
+    objs = APIToken.query(token=token)
+
+    # Authorization error
+    if not objs:
+        return get_json_result(
+            data=False, retmsg='Token is not valid!"', retcode=RetCode.AUTHENTICATION_ERROR)
+
+    tenant_id = objs[0].tenant_id
+    req = request.json
```

Comment:
use full name: request_body as the var name.

Reply:
Done